Modules/LearningSequence/classes/Player/class.ilObjLearningSequenceLearnerGUI.php

```diff
@@ -352,6 +360,7 @@ protected function getKioskRenderer(LSUrlBuilder $url_builder)
 
 			$this->kiosk_renderer = new ilKioskPageRenderer(
 				$this->tpl,
+				$this->layout_service,
```

Comment:
This is an antipattern that @maxbecker has brought up, I think he named it  `Courier Antipattern` (good naming, though). The thing is: The `ilObjLearningSequenceLearnerGUI` never uses the `LayoutServices` but only passes it on to the next class.

Make the `ilObjLearningSequenceLearnerGUI` depend on an `ilKioskPageRenderer` instead and move the construction and dependency resolution to the topmost layer accessible to us (which should be `ilObjLearningSequenceGUI`, imo).

In a second step we could then think of introducing a local DI, like we do for our plugins, but not in this PR here.

Reply:
Well, the dependencies are quite nested here; I totally agree on the need to change that, but I'd rather go directly for the DI.
We might take this in the meantime to have anything running available - but I'm also just as happy to close this PR.